Source/RequestInterceptor.swift

```diff
@@ -42,3 +54,8 @@ public protocol RequestRetrier {
     /// - parameter completion: The completion closure to be executed when retry decision has been determined.
     func should(_ manager: Session, retry request: Request, with error: Error, completion: @escaping RequestRetryCompletion)
 }
+
+// MARK: -
+
+/// A type that intercepts requests to potentially adapt and retry them.
+public protocol RequestInterceptor: RequestAdapter, RequestRetrier {}
```

Comment:
I thought it made the most sense to combine the `RequestAdapter`, `RequestRetrier`, and `RequestInterceptor` into a single file.